app/admin/contributions.rb

```diff
@@ -113,9 +113,13 @@ def approve
     end
 
     def refuse
-      resource.refuse!(current_user.id)
-      resource.update(rejected_reason: params["rejected_reason"])
-      redirect_back fallback_location: resource_path, notice: I18n.t('contribution_refused')
+      ActiveRecord::Base.transaction do
+        resource.refuse!(current_user.id)
+        resource.update!(rejected_reason: params["rejected_reason"])
+        redirect_back fallback_location: resource_path, notice: I18n.t('contribution_refused')
```

Comment:
why is the redirect_back inside the transaction?

Reply:
Oh, my mistake. Already changed